File: 프로그래머스/2/131704. 택배상자/택배상자.py
from collections import deque
def solution(order):
    q = deque()
    order = deque(order)
    answer = []
    for i in range(1,len(order)+1):
        q.append(i)
        while q:
            if q[-1] == order[0]:
                answer.append(q.pop())
                order.popleft()
                
            else: 
                break
    
    return len(answer)

File: 프로그래머스/2/131704. 택배상자/택배상자.cpp
#include <string>
#include <vector>
#include <stack>
#include <queue>
using namespace std;

int solution(vector<int> order) {
    int answer = 0;
    stack<int> st;
    queue<int> q;
    bool flag = true;
    for (int i = 1; i <= order.size(); i++){
        q.push(i);
    }
    
    for (int i = 0; i < order.size(); i++) {
        while (true) {
            if (order[i] == q.front()){
                answer++;
                q.pop();
                break;
            }
            else if (st.size()>0 && st.top() == order[i]){
                answer++;
                st.pop();
                break;
            }
            else if (q.size() == 0) {
                flag = false;
                break;
            }
            else {
                int tmp = q.front();
                q.pop();
                st.push(tmp);
            }
        }
        if (not flag) {
            break;
        }
    }    
    return answer;
}